climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,812 @@
+        parent-geometry.
+        If 'avg', the impact is averaged over all points in each geometry.
+        If 'sum', the impact is summed over all points in each geometry.
+        The default is 'sum'.
+
+    Returns
+    -------
+    Impact
+        Impact object with the impact per geometry (rows of exp.gdf). Contains
+        two additional attributes 'geom_exp' and 'coord_exp', the first one
+        being the origninal line or polygon geometries for which impact was
+        computed.
+
+    See Also
+    --------
+    exp_geom_to_pnt: disaggregate exposures
+
+    """
+
+    # disaggregate exposure
+    exp_pnt = exp_geom_to_pnt(
+        exp=exp, res=res,
+        to_meters=to_meters, disagg_met=disagg_met,
+        disagg_val=disagg_val
+        )
+    exp_pnt.assign_centroids(haz)
+
+    # compute point impact
+    impact_pnt = Impact()
+    impact_pnt.calc(exp_pnt, impf_set, haz, save_mat=True)
+
+    # re-aggregate impact to original exposure geometry
+    impact_agg = impact_pnt_agg(impact_pnt, exp_pnt, agg)
+
+    return impact_agg
+
+def impact_pnt_agg(impact_pnt, exp_pnt_gdf, agg):
+    """
+    Aggregate the impact per geometry.
+
+    The output Impact object contains an extra attribute 'geom_exp'
+    containing the geometries.
+
+    Parameters
+    ----------
+    impact_pnt : Impact
+        Impact object with impact per exposure point (lines of exp_pnt)
+    exp_pnt_gdf : GeoDataFrame
+        Geodataframe of an exposures featuring a multi-index. First level indicating
+        membership of original geometries, second level the disaggregated points.
+        The exposure is obtained for instance with the disaggregation method
+        exp_geom_to_pnt().
+    agg : string, optional
+        If 'agg', the impact is averaged over all points in each geometry.
+        If 'sum', the impact is summed over all points in each geometry.
+        The default is 'sum'.
+
+    Returns
+    -------
+    impact_agg : Impact
+        Impact object with the impact per original geometry. Original geometry
+        additionally stored in attribute 'geom_exp'; coord_exp contains only
+        representative points (lat/lon) of those geometries.
+
+    See also
+    --------
+        exp_geom_to_pnt: exposures disaggregation method
+    """
+
+    # aggregate impact
+    mat_agg = aggregate_impact_mat(impact_pnt, exp_pnt_gdf, agg)
+
+    # write to impact obj
+    impact_agg = set_imp_mat(impact_pnt, mat_agg)
+
+    # add exposure representation points as coordinates
+    repr_pnts = gpd.GeoSeries(
+        exp_pnt_gdf['geometry_orig'][:,0].apply(
+            lambda x: x.representative_point()))
+    impact_agg.coord_exp = np.array([repr_pnts.y, repr_pnts.x]).transpose()
+
+    # Add original geometries for plotting
+    impact_agg.geom_exp = exp_pnt_gdf.xs(0, level=1)\
+        .set_geometry('geometry_orig')\
+            .geometry.rename('geometry')
+
+    return impact_agg
+
+
```

Comment:
@emanuel-schmid I tried implementing this logic. Could you check that this is how it's done now in the full code?

Reply:
@Evelyn-M Thanks! 😃 I've amended one small change. When checking for the value of an enum the preferable way is to

```python
if AggMethod(agg_met) is AggMethod.SUM:
```

instead of

```python
if agg_met.value == 'sum':
```

With the former, there is a lot of checking done in the background, and the error message - if any - points directly to the problem: `ValueError: 'fix' is not a valid AggMethod`